src/Symfony/Component/Cache/Tests/Simple/FilesystemCacheTest.php

```diff
@@ -22,4 +22,52 @@ public function createSimpleCache($defaultLifetime = 0)
     {
         return new FilesystemCache('', $defaultLifetime);
     }
+
+    public function testPrune()
```

Comment:
Does this test fail without the other changes?
Looking at {{FilesystemCache::doContains}}, I think the cache will return false for expired entries, whether or not they have been pruned.